build-support/bin/cherry_pick.sh

```diff
@@ -25,16 +25,33 @@ if [[ -z $PR_NUM ]]; then
   read -r PR_NUM
 fi
 
-MILESTONE=$2
-if [[ -z $MILESTONE ]]; then
-  MILESTONE=$(gh pr view "$PR_NUM" --json milestone --jq '.milestone.title')
-  if [[ -z $MILESTONE ]]; then
+TARGET_MILESTONE=$2
+if [[ -z $TARGET_MILESTONE ]]; then
+  TARGET_MILESTONE=$(gh pr view "$PR_NUM" --json milestone --jq '.milestone.title')
+  if [[ -z $TARGET_MILESTONE ]]; then
     echo "No milestone on PR. What's the milestone? (E.g. 2.10.x)"
-    read -r MILESTONE
+    read -r TARGET_MILESTONE
   fi
 fi
 
+# NB: Find all milestones >= $TARGET_MILESTONE by having GH list them, then uses awk to trim the
```

Comment:
For the record, this much was clear all along. what wasn't clear was why that awk incantation achieves this. Assume that the reader understands piping and stuff, but doesn't know awk, because who does?

Reply:
Like 80% of any bash script is unintelligible symbols for me. I don't think we should be documenting how each line works. 🤷‍♂️

And it's kinda over-cliched at this point, but if you ask ChatGPT what it does it actually gives a nice summary 😉

Reply:
We expect people to have some basic bash familiarity, but awk is very rarely used and is especially unintelligible. And if we're relying on some non-obvious bashisms then I would expect to document those too.

Plus, if ChatGPT generated this then that makes me more nervous, not less... 

What is the argument against more documentation here? "You can always go futz with ChatGPT or RTFM" doesn't seem like the most reader-friendly approach?

Reply:
It just feels a bit cherry-picked (pun intended). We aren't explaining graphQL, or the `while IFS= read -r REVIEWER; do PR_CREATE_CMD+=(--reviewer "$REVIEWER"); done <<< "$REVIEWERS"` line (which I don't understand, but meh I get the intent and this is a script)